### PR TITLE
fix: use non hardcoded labels as it doesn't scale

### PR DESCRIPTION
**Pull request by sakulstra**
Soon there will be a new eMode added and the current id - name mapping doesn't scale very well as these groups could vary from market to market or network to network.
This pr uses the native eModeLabels

### CODE REVIEW

File: src/components/transactions/Emode/EmodeModalContent.tsx [hunk 1]
```diff
@@ -40,78 +43,43 @@ export interface EmodeModalContentProps {
...
-  const [selectedEmode, setSelectedEmode] = useState<EmodeCategory | undefined>(undefined);
-  const [emodeCategories, setEmodeCategories] = useState<Record<number, EmodeCategory>>({});
-
+  const [selectedEmode, setSelectedEmode] = useState<EmodeCategory | undefined>(
+    getInitialEmode(mode, eModes, user.userEmodeCategoryId)
+  );
   const networkConfig = getNetworkConfig(currentChainId);
 
-  // Create object of available emodes
-  useEffect(() => {
```
Comment (sakulstra) [inline src/components/transactions/Emode/EmodeModalContent.tsx:62]: this code was quite literally hell to read and the useEffect was kinda unnecessary

File: src/components/transactions/Emode/EmodeNaming.ts [hunk 2]
```diff
@@ -1,13 +1,6 @@
-export const getEmodeMessage = (categoryId: number, baseAssetSymbol: string): string => {
-  if (categoryId === 0) {
+export const getEmodeMessage = (label: string): string => {
```
Comment (sakulstra) [inline src/components/transactions/Emode/EmodeNaming.ts:1]: it's probably a bit stupid to keep the function, but was the "least effect" change

**Review by drewcook — APPROVED** [review]
These look to be some great optimizations. Everything checks out in the demo URL also 👍

**Review by defispartan — APPROVED** [review]
Looks good, much cleaner

**Pull request merged**